app/javascript/mastodon/actions/streaming.js

```diff
@@ -132,21 +131,17 @@ export const connectTimelineStream = (timelineId, channelName, params = {}, opti
 
 /**
  * @param {Function} dispatch
- * @param {function(): void} done
  */
-const refreshHomeTimelineAndNotification = (dispatch, done) => {
-  // @ts-expect-error
-  dispatch(expandHomeTimeline({}, () =>
-    // @ts-expect-error
-    dispatch(expandNotifications({}, () =>
-      dispatch(fetchAnnouncements(done))))));
-};
+async function refreshHomeTimelineAndNotification(dispatch) {
+  await dispatch(expandHomeTimeline({ maxId: undefined }));
```

Comment:
A difference in behaviour here is that if any of those promise action fail, then I think it will throw an exception, which is not caught here, and the next ones will not be executed?

Reply:
I had not tested, but I just tested by adding some delay as well as stopping the server, the queries are done sequentially and don't stop on error, just like before.